src/middlewares/validateReturnTrip.js

```diff
@@ -1,96 +1,109 @@
-    
-    const isTravelDateValid = async (travelDate) => {
-      const todayDate = new Date(getTodayDate());
-      travelDate = new Date(travelDate);
-      
-      if(travelDate >= todayDate){
-        //check whether travelDate is set after the prev requested trip.
-        const latestRequest = await models.Request.findAll({
+   const urlSections = req.urlPathSections;
+   if(urlSections)
+    address = urlSections[urlSections.length -2 ] || null;
+   
+   if(address === 'edit') {
+    const isRequestEmpty = isObjectEmpty(req.body);    
+    if(isRequestEmpty === true) throw('Empty request');
+    else schema = editRequestSchema;
+    } else schema = createTwoWayTripSchema;
+   
+   const { error } = schema.validate({ origin, destination, departureDate, returnDate, reason, accommodation});
+   
+    if (error) throw error;
+    else if(origin === destination) throw 'similar origin and destination';
+    else if(reqReturnDate < reqDepartureDate) throw 'departureDate > returnDate' ;
+    else if(reqDepartureDate < todayDate) throw 'past departure date' ; 
+    else if(departureDate || returnDate){
+    const requests = await models.Request.findAll({
           where : {
-            requesterId: id
-          },
-          limit: 1,
-          order: [ [ 'returnDate', 'DESC' ]] ,    
-          raw: true
-         })         
-         let latestRequestDate;
-         if(!latestRequest.length) return true;
-         else {           
-           latestRequestDate = new Date(latestRequest[0].departureDate);
-           if(travelDate > latestRequestDate) return true;
-           else return false;
-         }
-      } else return false;
-    }
-  const { error } = schema.validate({ origin, destination, departureDate, returnDate, reason, accommodation});
-  if (error) {
-    return res.status(422).json({
-      status: res.statusCode,
-      error: error.message,
-    });
+            requesterId: req.user.id
+          },raw: true
+        })  
+        let totalRequests = requests.length;
 
-  } else{
-    const condition1 = isOriginDiffersFromDestination(origin,destination);
-    const condition2 = isReturnDateValid(departureDate,returnDate);
-    const condition3 = await isTravelDateValid(departureDate);
-    if(condition1 === true && condition2 === true && condition3 === false){
-      return res.status(422).json({
-        status: res.statusCode,
-        error: "Departure date should be chosen from today's date and has to not collide with your previous requested trip!",
-      });
-    }
-    if(condition1 === true && condition3 === true && condition2 === false){
-      return res.status(422).json({
-        status: res.statusCode,
-        error: "Returning date has to be the day after your departure's date!",
-      });
-    }
-    if(condition2 === true && condition3 === true && condition1 === false){
+      for(let index = 0; index < totalRequests; index ++){
```

Comment:
if you can replace this for loop with forEach it could be better

Reply:
@EugeneMunya, considering the performance, the for-loop is `faster` than for-Each and the forEach is only terminated whenever all elements are iterated unlike the for-loop.
That is the main reasons I prefer for-loop than the for-each

Reply:
Hello @T2Wil, I think you can yous map?